src/item_location.cpp

```diff
@@ -189,6 +299,34 @@ class item_location::item_on_vehicle : public item_location::impl {
         what = nullptr;
     }
 
+    std::string describe( const Character *ch ) const override {
+        std::string res = veh->parts[partnum].info().name;
+        if( ch ) {
+            ; // @todo implement relative decriptions
+        }
+        return res;
+    }
+
+    int obtain( Character& ch ) override {
+        if( !what ) {
+            return INT_MIN;
+        }
+
+        int mv = 0;
+
+        // @todo handle unpacking costs
+        // @todo account for distance
+
+        mv += dynamic_cast<player *>( &ch )->item_handling_cost( *what );
```

Comment:
Btw. you can dynamic cast reference, no need to convert the value to a pointer first.